packages/driver-adapter-utils/src/binder.ts

```diff
@@ -26,11 +28,25 @@ class ErrorRegistryInternal implements ErrorRegistry {
   }
 }
 
-// *.bind(adapter) is required to preserve the `this` context of functions whose
-// execution is delegated to napi.rs.
-export const bindAdapter = (adapter: SqlDriverAdapter): ErrorCapturingSqlDriverAdapter => {
+export const bindSqlAdapterFactory = (adapterFactory: SqlDriverAdapterFactory): ErrorCapturingSqlDriverAdapterFactory => {
   const errorRegistry = new ErrorRegistryInternal()
 
+  const boundFactory: ErrorCapturingSqlDriverAdapterFactory = {
```

Comment:
```suggestion
  const boundFactory: ErrorCapturingSqlDriverAdapterFactory = {
```
won't we also want the equivalent for `SqlMigrationAwareDriverAdapterFactory`

Reply:
Not right now, as most driver adapters don't currently support it anyway.